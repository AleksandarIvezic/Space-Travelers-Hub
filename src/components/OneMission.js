import React from 'react';
import PropTypes from 'prop-types';
import { joinMission } from '../redux/missions/missions';
import store from '../redux/storeConfig';

const OneMission = (props) => {
  const { name, description, id } = props;

  const handleJoin = (e) => {
    store.dispatch(joinMission(e.target.id));
  };

  return (
    <>
      <th scope="row">{name}</th>
      <td>{description}</td>
      <td>
        <span className="badge bg-secondary">NOT A MEMBER</span>
      </td>
      <td className="text-center">
        <button
          type="button"
          className="btn btn-sm btn-outline-secondary"
          onClick={handleJoin}
          id={id}
        >
          JoinMission
        </button>
      </td>
    </>
  );
};

OneMission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default OneMission;
