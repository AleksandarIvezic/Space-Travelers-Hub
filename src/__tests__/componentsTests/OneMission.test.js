import React from 'react';
import renderer from 'react-test-renderer';
import OneMission from '../../components/OneMission';

describe('OneMission', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<OneMission reserved={false}
        id={'123'}
        name={'name'}
        description={'desc'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
