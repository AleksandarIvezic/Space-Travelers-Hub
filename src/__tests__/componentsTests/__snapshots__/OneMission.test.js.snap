// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneMission renders correctly 1`] = `
Array [
  <th
    scope="row"
  >
    name
  </th>,
  <td>
    <div
      className="descri"
    >
      desc
    </div>
  </td>,
  <td>
    <span
      className="badge bg-secondary"
    >
      NOT A MEMBER
    </span>
  </td>,
  <td
    className="text-center"
  >
    <button
      className="btn btn-sm btn-outline-secondary mission-btn"
      id="123"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>,
]
`;
